app/assets/javascripts/discourse/models/user-stream.js.es6

```diff
@@ -30,14 +30,16 @@ export default RestModel.extend({
     "/user_actions.json?offset=%@&username=%@"
   ),
 
-  filterBy(filter, noContentHelpKey) {
+  filterBy(filter, noContentHelpKey, actingUsername) {
```

Comment:
This API could be a good candidate for using object literals for the options. I always think of the call site:

`filterBy(123, 'some_default_key', user.username)`

In this case I can still somewhat understand it but I almost always prefer:

`filterBy(123, { defaultKey: 'some_key, actingUsername: user.username })`.

Reply:
I feel that it makes the resulting function harder to understand considering that the function is expecting 3 exact inputs.

`filterBy(filter, noContentHelpKey, actingUsername)`

vs

`filterBy(filter, options)`

Since the latter requires us to read through the function to figure out the keys in the options object is going to be, I find it less readable then before.

Reply:
I see what you mean, but you aren't using `noContentHelpKey` or `actingUsername` in any meaningful way here. You are passing them through to `setProperties`.

It would be simpler and less code if you took an argument called `props`. You could either merge it into the literal before calling `setProperties` or could call `setProperties` a second time with the props.

Reply:
Followed up in https://github.com/discourse/discourse/commit/40533bc570685e3102f2df81d92d7a3c52f9bcb8.